src/Shaders/ShadersInclude/samplerVertexImplementation.fx

```diff
@@ -0,0 +1,26 @@
+#if defined(_DEFINENAME_) && _DEFINENAME_DIRECTUV == 0
+	if (v_INFONAME_ == 0.)
+	{
+		v_VARYINGNAME_UV = vec2(_MATRIXNAME_Matrix * vec4(uvUpdated, 1.0, 0.0));
+	}

```

Comment:
Could we add #ifdef around them as we know the UVx from the #define ?

Reply:
on all but the first one :-)

Reply:
What do you mean? I don't understand, can you give an example?

Reply:
```suggestion
#ifdef UV1
	else if (v_INFONAME_ == 1.)
	{
		v_VARYINGNAME_UV = vec2(_MATRIXNAME_Matrix * vec4(uv2, 1.0, 0.0));
	}
#endif
#ifdef UV2
	else if (v_INFONAME_ == 2.)
	{
		v_VARYINGNAME_UV = vec2(_MATRIXNAME_Matrix * vec4(uv3, 1.0, 0.0));
	}
#endif
#ifdef UV3
	else if (v_INFONAME_ == 3.)
	{
		v_VARYINGNAME_UV = vec2(_MATRIXNAME_Matrix * vec4(uv4, 1.0, 0.0));
	}
#endif
```

Reply:
wondering if that makes sense ?